### DISCUSSION TITLE
How to get object properties and count in same query

### DISCUSSION THREAD

**Discussion by milestones95** [supabase/supabase#2786] [Questions] [answered]
Hi is it possible for me to query all the list of writers I have in my db, retrieve their properties, and also get the total count of stories they have that should be mapped to their writer_id? Can I do this in one query? Here's what I have so far:

```sql
create or replace function "ListAllWriters"()
returns table(writer_id uuid, color varchar, username varchar, about varchar, price float, country varchar) as $$
  select distinct
    wi.writer_id, u.color_code, u.username, wi.about, wi.price, wi.country
  from writer_info wi
  join users u on u.id = wi.writer_id
  join writer_payments wp on wp.user_id = wi.writer_id
  where wp."isPayoutEnabled" = true
$$ language sql stable;

```

**Comment by steve-chavez** [comment, marked as answer]
Hey @milestones95,

[rpc](https://supabase.io/docs/reference/javascript/rpc) supports the `count` parameter, so you should be able to get the count without modifying your query.

```js
supabase.rpc('ListAllWriters', {}, { count: 'exact' })
```

**Reply by milestones95** [reply]
I think I should have added more context here. I have a separate table called writer_requests and the properties are:
-fulfilled<boolean>
-writer_id<uuid>
-details<varchar>

How can I get the number of writer requests for a particular writer as I'm listing out all of the writers in the first query I referenced? Can I do them both in one single query?

**Reply by steve-chavez** [reply]
@milestones95 A [postgreSQL CTE](https://www.postgresqltutorial.com/postgresql-cte/) might help if you want do it in pure SQL.
